wordpress-multi-server.yaml

```diff
@@ -359,16 +371,237 @@ resources:
       name: { get_param: "OS::stack_id" }
       save_private_key: true
 
-  # Server resources
+  wait_condition_secondaries:
+    type: OS::Heat::SwiftSignal
+    properties:
+      handle: { get_resource: wait_condition_handle_secondaries }
+      count: { get_param: wp_web_server_count }
+      timeout: 1800
+
+  wait_condition_handle_secondaries:
+    type: OS::Heat::SwiftSignalHandle
+
+  database_server:
+    type: OS::Trove::Instance
+    properties:
+      name:
+        str_replace:
+          template: "wordpress_db_%stackid%"
+          params:
+            "%stackid%": { get_param: "OS::stack_id" }
+      flavor: { get_param: db_flavor }
+      size: { get_param: db_size }
+      datastore_type: mysql
+      datastore_version: "5.6"
+      databases:
+      - name: { get_param: database_name }
+      users:
+      - name: { get_param: username }
+        password: { get_resource: database_password }
+        databases: [ { get_param: database_name } ]
+
+  wait_condition_primary:
+    type: OS::Heat::SwiftSignal
+    properties:
+      handle: { get_resource: wait_condition_primary_handle }
+      count: 1
+      timeout: 1800
+
+  wait_condition_primary_handle:
+    type: OS::Heat::SwiftSignalHandle
+
+  wp_web_servers:
+    type: "OS::Heat::ResourceGroup"
+    depends_on: database_server
+    properties:
+      count: { get_param: wp_web_server_count }
+      resource_def:
+        type: "http://drop.duncancreek.net/wordpress-web-server.yaml"
+        properties:
+          wp_web_server_hostnames: { get_param: wp_web_server_hostnames }
+          wp_web_server_flavor: { get_param: wp_web_server_flavor }
+          image: { get_param: image }
+          ssh_keypair_name: { get_resource: ssh_key }
+          wait_condition_handle_secondaries_curl_cli: { get_attr: ['wait_condition_handle_secondaries', 'curl_cli'] }
+          domain: { get_param: domain }
+          mysql_root_password: { get_attr: [mysql_root_password, value] }
+          database_name: { get_param: database_name }
+          database_server: { get_attr: [database_server, hostname]}
+          database_password: { get_resource: database_password }
+
+  wordpress_primary_cloud_init:
+    depends_on: wp_web_servers
```

Comment:
No need for depends on here. This resource only creates the content. It doesn't deploy anything.